tests/Operation/DeleteFunctionalTest.php

```diff
@@ -63,6 +64,46 @@ public function testDeleteMany(): void
         $this->assertSameDocuments($expected, $this->collection->find());
     }
 
+    public function testHintOptionUnsupportedClientSideError(): void
```

Comment:
Are these scenarios not covered by the automated spec tests, or did you want to ensure that the correct exception thrown?

Reply:
Had I not implemented [PHPLIB-711](https://jira.mongodb.org/browse/PHPLIB-711), we could rely on the spec tests to differentiate between PHPLIB's UnsupportedException and the BulkWriteException thrown from PHPC encountering libmongoc's client-side error. However, I think unwrapping BulkWriteExceptions is necessary to properly differentiate cilent-side and server-side errors.

With PHPLIB-711 implemented, the unwrapped BulkWriteException (in this particular case) and UnsupportedException are both considered client-side errors so the spec tests really cannot differentiate. That's where these tests come in, to explicitly assert that PHPLIB raises its own internal exception.